crystal_diffusion/models/score_network.py

```diff
@@ -199,3 +215,112 @@ def _forward_unchecked(self, batch: Dict[AnyStr, torch.Tensor]) -> torch.Tensor:
 
         output = self.mlp_layers(input).reshape(positions.shape)
         return output
+
+
+@dataclass(kw_only=True)
+class MACEScoreNetworkParameters(BaseScoreNetworkParameters):
+    """Specific Hyper-parameters for MACE score networks."""
+
+    number_of_atoms: int  # the number of atoms in a configuration.
+    r_max: float = 5.0
+    num_bessel: int = 8
+    num_polynomial_cutoff: int = 5
+    max_ell: int = 2
+    interaction_cls: str = "RealAgnosticResidualInteractionBlock"
+    interaction_cls_first: str = "RealAgnosticInteractionBlock"
+    num_interactions: int = 2
+    hidden_irreps: str = "128x0e + 128x1o"  # irreps for hidden node states
+    MLP_irreps: str = "16x0e"  # from mace.tools.arg_parser
+    atomic_energies: np.ndarray = field(default_factory=lambda: np.zeros((1,)))
+    avg_num_neighbors: int = 1  # normalization factor for the message
+    correlation: int = 3
+    gate: str = "silu"  # non linearity for last readout - choices: ["silu", "tanh", "abs", "None"]
+    radial_MLP: str = "[64, 64, 64]"  # "width of the radial MLP"
+    radial_type: str = "bessel"  # type of radial basis functions - choices=["bessel", "gaussian", "chebyshev"]
+    n_hidden_dimensions: int  # the number of hidden layers for the final MLP - should be small
+    hidden_dimensions_size: int  # the dimensions of the hidden layers - should be small
+
+
+class MACEScoreNetwork(ScoreNetwork):
+    """Score network using atom features from MACE.
+
+    Inherits from the given framework's model class.
+    """
+
+    def __init__(self, hyper_params: MACEScoreNetworkParameters):
+        """__init__.
+
+        Args:
+            hyper_params (dict): hyper parameters from the config file.
```

Comment:
No need to mention the type of `hyper_params` since it's already in the method signature. Also, it's not a dict...